### PR TITLE
fix: `getOperation` for `Transfer Asset`

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close https://github.com/FuelLabs/fuels-ts/issues/1411
- Close https://github.com/FuelLabs/fuels-ts/issues/1626

### CODE REVIEW

File: packages/providers/src/transaction-summary/operations.test.ts [hunk 1]
```diff
@@ -795,7 +817,7 @@ describe('operations', () => {
         OPERATION_CONTRACT_CALL.assetsSent[0].assetId
       );
     });
-    it('should stack when same asset is added together with a different asset', () => {
+    it('should stack when same asset is added together with a different asset [CONTRACT-CALL]', () => {
```
Comment (arboleya) [inline packages/providers/src/transaction-summary/operations.test.ts:820]: Nit / line-break.

```suggestion

    it('should stack when same asset is added together with a different asset [CONTRACT-CALL]', () => {
```

File: packages/providers/src/transaction-summary/operations.test.ts [hunk 2]
```diff
@@ -824,6 +846,105 @@ describe('operations', () => {
       );
       expect(operationsAddedSameAsset[0].assetsSent?.[1]?.assetId).toEqual(DIF_ASSET_ID);
     });
+    it('ensure operation asset transfer stacks multiple assetSents between same addresses', () => {
```
Comment (arboleya) [inline packages/providers/src/transaction-summary/operations.test.ts:849]: Nit / line-break.

```suggestion

    it('ensure operation asset transfer stacks multiple assetSents between same addresses', () => {
```

File: packages/providers/src/transaction-summary/operations.test.ts [hunk 3]
```diff
@@ -824,6 +846,105 @@ describe('operations', () => {
...
+        operationOne.assetsSent?.[0].assetId
+      );
+      expect(stackedOperation[0].assetsSent?.[1]?.amount.valueOf()).toEqual(
+        operationTwo.assetsSent?.[0]?.amount.valueOf()
+      );
+      expect(stackedOperation[0].assetsSent?.[1]?.assetId).toEqual(
+        operationTwo.assetsSent?.[0].assetId
+      );
+    });
+    it('ensure operation asset transfer does not stack multiple assetSents between different addresses', () => {
```
Comment (arboleya) [inline packages/providers/src/transaction-summary/operations.test.ts:888]: Nit / line-break.

```suggestion

    it('ensure operation asset transfer does not stack multiple assetSents between different addresses', () => {
```

File: packages/providers/src/transaction-summary/operations.test.ts [hunk 4]
```diff
@@ -824,6 +846,105 @@ describe('operations', () => {
...
+        operationOne.assetsSent?.[0].assetId
+      );
+      expect(multipleOperations[1].assetsSent?.length).toEqual(1);
+      expect(multipleOperations[1].assetsSent?.[0]?.amount.valueOf()).toEqual(
+        operationTwo.assetsSent?.[0]?.amount.valueOf()
+      );
+      expect(multipleOperations[1].assetsSent?.[0]?.assetId).toEqual(
+        operationTwo.assetsSent?.[0].assetId
+      );
+    });
```
Comment (arboleya) [inline packages/providers/src/transaction-summary/operations.test.ts:947]: Nit / line-break.

```suggestion
    });
 
```

**Review by arboleya — COMMENTED** [review]
Is there any reason why you removed all comments from the source?

I'm not saying the previous comments were 100% clear, but having no comments at all is probably worse.

File: packages/providers/src/transaction-summary/operations.ts [hunk 5]
```diff
@@ -128,43 +139,37 @@ function isSameOperation(a: Operation, b: Operation) {
...
+    allOperations[index] = existentOperation;
+  } else {
+    // New operation, we can simply add it.
+    allOperations.push(toAdd);
+  }
 
-  // if this operation didn't exist before just add it to the end
-  return ops.length ? ops : [...operations, toAdd];
+  return allOperations;
 }
```
Comment (nedsalk) [inline packages/providers/src/transaction-summary/operations.ts:173]: Here's another take on it. After looking into the code, IMO the comments are redundant as they're just explaining control flow and aren't giving any additional context.

```suggestion
/** @hidden */
export function addOperation(operations: Operation[], toAdd: Operation) {
  const index = operations.findIndex((op) => isSameOperation(op, toAdd));

  if (index === -1) {
    return [...operations, toAdd];
  }

  const operation = operations[index];

  const updatedOperation = {
    ...operation,
    assetsSent: mergeAssets(operation, toAdd),
    calls: [...(operation.calls || []), ...(toAdd.calls ?? [])],
  };

  return [
    ...operations.slice(0, index),
    updatedOperation,
    ...operations.slice(index + 1, operations.length),
  ];
}
```

**Review by nedsalk — DISMISSED** [review]

**Review by danielbate — APPROVED** [review]

**Comment by danielbate** [comment]
Approved, however we do not have a changeset for this PR but CI is passing. Are we doing changesets manually for `rc/salamander` @arboleya ?

**Comment by danielbate** [comment]
Not tested but I bet that is the case. Shall we set the [baseBranch config](https://github.com/changesets/changesets/blob/main/docs/config-file-options.md)?

**Review by nedsalk — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**